.github/scripts/validateImmutableActionRefs.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+############################################
+#    Check for unsafe action references    #
+############################################
+
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+readonly SCRIPT_DIR
+GITHUB_DIR="$(dirname "$SCRIPT_DIR")"
+readonly GITHUB_DIR
+
+source "$SCRIPT_DIR/shellUtils.sh"
+
+title 'Checking for mutable action references...'
+
+ACTION_USAGES=''
+
+# Find yaml files in the `.github` directory
+# shellcheck disable=SC2044
+for FILE in $(find "$GITHUB_DIR" -type f \( -name "*.yml" -o -name "*.yaml" \)) ; do
+    USES_LINES="$(grep --no-filename 'uses:' "$FILE")"
+
+    # Ignore files without external action usages
+    if [[ -z "$USES_LINES" ]]; then
+        continue
+    fi
+
+    # Normalize: remove leading -
+    USES_LINES="${USES_LINES//- uses:/uses:}"
+
+    # Normalize: remove quotes
+    USES_LINES="${USES_LINES//\"/ }"
+    USES_LINES="${USES_LINES//\'/ }"
+
+    # Normalize: remove carriage returns
+    USES_LINES="${USES_LINES//\\r/ }"
+
+    # Grab action names and refs from uses lines.
+    # At this point, these lines look like "uses: myAction@ref", so `awk '{print $2}'` just grabs the second word from each line.
+    ACTION_USAGES+=$'\n'
+    ACTION_USAGES+="$(echo "$USES_LINES" | awk '{print $2}')"
+done
+
+# De-dupe and sort action usages
+ACTION_USAGES="$(echo "$ACTION_USAGES" | grep -vE '^$' | sort | uniq)"
+
+info 'All action usages...'
+echo "$ACTION_USAGES"
+echo
+
+# Ignore any local action usages, callable workflows, or Expensify-owned actions
+ACTION_USAGES="$(echo "$ACTION_USAGES" | grep -vE "^((./)?.github|Expensify/)")"
```

Comment:
Can you include some examples in the comment of what would be excluded here?

Reply:
done